Dalamud/Plugin/Services/IGameInteropProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Diagnostics;
+
+using Dalamud.Hooking;
+using Dalamud.Utility.Signatures;
+
+namespace Dalamud.Plugin.Services;
+
+/// <summary>
+/// Service responsible for the creation of hooks.
+/// </summary>
+public interface IGameInteropProvider
+{
+    /// <summary>
+    /// Available hooking backends.
+    /// </summary>
+    public enum HookBackend
+    {
+        /// <summary>
+        /// Choose the best backend automatically.
+        /// </summary>
+        Automatic,
+        
+        /// <summary>
+        /// Use Reloaded hooks.
+        /// </summary>
+        Reloaded,
+        
+        /// <summary>
+        /// Use MinHook.
+        /// You should never have to use this without talking to us first.
+        /// </summary>
+        MinHook,
+    }
+    
+    /// <summary>
+    /// Initialize <see cref="Hook{T}"/> members decorated with the <see cref="SignatureAttribute"/>.
```

Comment:
Explicitly specify that this also will initialize any delegates that are defined using the `[Signature]` attribute as well. At present, wording can be interpreted that this only does hooks.